### ISSUE TITLE
ISBAT: Switch between users when authenticating and keeping sessions for each user identifiable.

### ISSUE THREAD

**Issue by seb86** [co-cart/co-cart#107]
Need help identifying and correctly setting the user when authenticating using the REST API. This will allow for developers to switch between users/customers for testing purposes and keeping the sessions as individual sessions, not merged with others.

What we are trying to achieve with this issue is similar to the [User Switching](https://wordpress.org/plugins/user-switching/) plugin by [@johnbillion](https://github.com/johnbillion)

### Task

- [ ] Identify logged in user and set user ID in session.
- [ ] If user ID is set in session, validate it against the authentication cookie created by [`wp_set_auth_cookie()`](https://developer.wordpress.org/reference/functions/wp_set_auth_cookie/). If it does not match, set new user ID in session.
- [ ] If user ID in session has changed, then we must forget the current session using `WC()->session->forget_session();` before WooCommerce loads or creates a new session.
- [ ] If the user ID shows that the user is not authenticating, logout the user using [`wp_logout()`](https://developer.wordpress.org/reference/functions/wp_logout/), this will clear all authentication, cookies and sessions both WordPress and WooCommerce.

I assume that all of this checking will need to be done before any of the REST API controllers for CoCart are included say [here](https://github.com/co-cart/co-cart/blob/master/includes/class-cocart-init.php#L154), just before the WooCommerce session is initialized, but I could be wrong.

This task is a low priority at this time but one that would need considerable planning and testing.

If you would like to contribute your time for this issue or like to discuss the issue further and have ideas, please comment.

Thank you for reading.

Sébastien.

😄

**Comment by seb86** [comment]
This should be much easier to work on once v2.1 is released.

**Comment by seb86** [comment]
With v2.1 of CoCart, create two customers and add different items to the cart for each customer. Best to test one customer in Postman and the other customer via your app.

### Steps

1. Authenticate the first customer and add an item and view cart.
2. Authenticate the second customer and add an item without clearing the cart and view cart. 

The carts should not be the same or merged.

**Comment by seb86** [comment]
Solved. New documentation to write.
